src/database/operations/generators/generateCreateOrUpdateItemOperation.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Generates a function that will invoke a specific query for data insertion
+ *
+ * @template T - The type of the item being created.
+ * @template I - The type of object to be passed for data insertion.
+ *
+ * @param {string} queryName - The name of the query to execute.
+ * @param {Object} savedAttributes - A list of the attribute names which should be converted to query parameters.
+ *
+ * @returns {Function} A function that takes query parameters, limit, and offset, and returns a promise that resolves to a bundle of entries and total count.
+ */
```

Comment:
This comment should be moved to just above the function it describes so that IDEs can make use of it.